specs-go/v1/layout.go

```diff
@@ -14,8 +14,15 @@
 
 package v1
 
+import "regexp"
+
 // ImageLayout is the structure in the "oci-layout" file, found in the root
 // of an OCI Image-layout directory.
 type ImageLayout struct {
 	Version string `json:"imageLayoutVersion"`
 }
+
+var (
+	// RefsRegexp for string ref should be complied with description "Object names in the refs subdirectories MUST NOT include characters outside of the set of "A" to "Z", "a" to "z", "0" to "9", the hyphen -, the dot ., and the underscore _." at https://github.com/opencontainers/image-spec/blob/master/image-layout.md.
+	RefsRegexp = regexp.MustCompile(`^[a-zA-Z0-9\-\.\_]+?$`)
```

Comment:
You don't have to backslash-escape hyphen, period, or underscore inside your brackets.  And I'm not sure why you have the `?`.  So I think this should be something like:

```go
RefsRegexp = regexp.MustCompile(`^[a-zA-Z0-9-._]+$`)
```